lang/LangPrimSource/PyrFilePrim.cpp

```diff
@@ -162,11 +162,8 @@ int prFileMkDir(struct VMGlobals * g, int numArgsPushed)
 	if (error != errNone)
 		return error;
 
-	boost::system::error_code error_code;
 	const bfs::path& p = SC_Codecvt::utf8_str_to_path(filename);
-	bfs::create_directories(p, error_code);
-	if (error_code)
-		postfl("Warning: %s (\"%s\")\n", error_code.message().c_str(), p.c_str());
+	bfs::create_directories(p);
```

Comment:
per docs this function also returns a boolean indicating whether or not a new directory was created as a result of this call. It would be useful to return that as well, no?

Reply:
Not exactly. I haven't changed the function being called: it is still `create_directories()`, not `create_directory()`. Doc says:

> Returns: The value of `!exists(p)` prior to the establishment of the postcondition.

In other words, it distinguishes between returning successfully after creating a directory that didn't exist before, vs returning successfully with no action (because the directory already existed).

If users really need to know, they can already check `File.exists` before calling `File.mkdir` -- so it would be the difference between:

```
if(File.exists(dir).not) {
	try {
		File.mkdir(dir)
	} { |error|
		error.reportError
	}
} {
	"% already existed, no need to re-create\n".postf(dir.basename)
};

// vs
try {
	if(File.mkdir(dir).not) {
		"% already existed, no need to re-create\n".postf(dir.basename)
	}
} { |error|
	error.reportError
}
```

... the latter of which is admittedly slightly less verbose, but also seems to me less clear. Burying the crucial action in an `if` condition is not exactly good for legibility.

I can't see any other benefit. All other cases I can imagine where the directory would not be created end with an error being thrown.

Reply:
Well, I tried this:

```
int prFileMkDir(struct VMGlobals * g, int numArgsPushed)
{
	PyrSlot *b = g->sp;
	char filename[PATH_MAX];

	int error = slotStrVal(b, filename, PATH_MAX);
	if (error != errNone)
		return error;

	const bfs::path& p = SC_Codecvt::utf8_str_to_path(filename);
	bool result = bfs::create_directories(p);

	SetBool(g->sp, result);
	return errNone;
}
```

But `File.mkdir(aString)` still returns the File class, not the thing that I supposedly `SetBool`-ed.

I'm out of time. Chinese universities' spring semester is a lot later than American ones' -- I still have two exams to arrange and a pile of audio files to grade. Feel free to push a working change to get the Boolean result -- I don't object to the idea as such, but I'm not going to do any more on this myself.

Reply:
OK. I realized overnight that I could check the Writing Primitives documentation and find out where to set the return value. Done.

Note that I'm copying the style `PyrSlot *a = g->sp - 1, *b = g->sp;` from the other file primitives. If this is stylistically incorrect, that would involve changes outside the scope of this PR. So I think I've addressed all the concerns now.